content/docs/guides/build/bring-your-own-llvm.md

```diff
@@ -0,0 +1,125 @@
+---
+title: "Build with system-installed LLVM"
+weight: 3
+description: >
+  How to build TinyGo with a system-installed version of LLVM.
+---
+
+⚠️ Halt! This is the system installed LLVM guide! Please check the following table and make sure you don't need the [manual LLVM install guide](../manual-llvm) instead! 
+
+
+| You need to build LLVM manually in the following cases |
+|---|
+|  You would like to use it for the ESP8266 or ESP32 chips |
+| You are using Windows. |
+| Your Linux distribution (if you use Linux) does not ship the right LLVM version. |
+
+## System installed LLVM instructions
+
+### Installing LLVM
+
+See [troubleshooting](#troubleshooting) if running into issues during the instructions.
+
+Using a system-installed version of LLVM depends on your system, of course.
+
+#### Debian or Ubuntu
+
+For **Debian** or **Ubuntu** you can install LLVM by adding a new apt repository. For more information about this method, see [apt.llvm.org](https://apt.llvm.org/). *Before copying the command below, please replace `xxxxx` with your distribution's codename*.
+
+| Distro | Version | Codename |
+|--------|------- |-----------|
+| Ubuntu | 18.04  | `bionic`  |
+| Ubuntu | 20.04  | `focal`   |
+| Ubuntu | 20.10  | `groovy`  |
+| Ubuntu | 21.04  | `hirsute` |
+| Ubuntu | 22.04  | `jammy`   |
+| Debian | 10     | `buster`  |
+| Debian | 11     | `bullseye`|
+| Debian | sid    | `unstable`|
+
+```shell
+echo 'deb http://apt.llvm.org/xxxxx/ llvm-toolchain-xxxxx-16 main' | sudo tee /etc/apt/sources.list.d/llvm.list
+```
+
+After adding the apt repository for your distribution you may install the LLVM toolchain packages:
+
+```shell
+wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+sudo apt-get update
+sudo apt-get install clang-16 llvm-16-dev lld-16 libclang-16-dev
```

Comment:
We've switched to LLVM 17 in the current dev branch. We will need a separate PR to replace all LLVM 16 references to LLVM 17.